### PR TITLE
[11.0] [MIG] report_qweb_pdf_watermark

### PR DESCRIPTION
**Pull request by schout-it**
migration, now with module history

### CODE REVIEW

**Comment by schout-it** [comment]
Had a look at the test errors, but it seems that when the test cursor is used, render_qweb_pdf returns the html version of the report.
Which renders the test useless, as _run_wkhtmltopdf is never called

**Comment by pedrobaeza** [comment]
If you look at the code:

https://github.com/odoo/odoo/blob/750e70e0a12c11d50998a44dff447f07fd7d06c3/odoo/addons/base/ir/ir_actions_report.py#L566-L567

what is preventing the PDF generation is the absence of a `test_report_directory` configuration variable.

@hbrunn I remember you trick this variable in one .travis.yml file, isn't it? Can you point us how to do it?

**Comment by schout-it** [comment]
That is one part, but a little further there is a check for the cursor type which will also block:
https://github.com/odoo/odoo/blob/750e70e0a12c11d50998a44dff447f07fd7d06c3/odoo/addons/base/ir/ir_actions_report.py#L593-L594

**Comment by pedrobaeza** [comment]
Ah, that new one is good for preventing deadlocks as said, but then I don't know how to do it...

**Comment by pedrobaeza** [comment]
I have found how to do it (at least the directory part): https://github.com/OCA/edi/blob/11.0/.travis.yml#L42

Please try.

**Comment by pedrobaeza** [comment]
Now maybe you should add also wkhtmltopdf: https://github.com/OCA/edi/blob/11.0/.travis.yml#L26

**Comment by schout-it** [comment]
wkhtml2pdf added, and also added pip dependency for PyPDF2

**Review by pedrobaeza — APPROVED** [review]
Squashing the 11.0 commit history before merging.

File: report_qweb_pdf_watermark/README.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+To use this module, you need to:
+
+#. go to your report
+#. select a PDF or image to use as watermark. Note that resolutions and size must match, otherwise you'll have funny results
+#. You can also fill in an expression that returns the data (base64 encoded) to be used as watermark
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/143/8.0
```
Comment (LoisRForgeFlow) [inline report_qweb_pdf_watermark/README.rst:29]: 11.0

Comment (schout-it) [reply report_qweb_pdf_watermark/README.rst:29]: seems even with 11.0 I can't test the module.
What should I change the 143 to, to get the right runbot?

Comment (schout-it) [reply report_qweb_pdf_watermark/README.rst:29]: answering my own question, as long as this is not merged, it will not show up on that runbot

**Comment by pedrobaeza** [comment]
Now you can try on runbot

**Comment by ovnicraft** [comment]
@schout-it please resolve conflicts

**Comment by MiquelRForgeFlow** [comment]
You should do a correct rebase.

**Comment by schout-it** [comment]
had some trouble rebasing, did a diff at the end and fixed what was still wrong

**Comment by MiquelRForgeFlow** [comment]
please do
```
    git fetch [remote] 11.0
    git rebase -i [remote]/11.0
```

where `[remote]` should be `origin` (if `origin` points to OCA repo) or `OCA` or `other_name_you_use_for_OCA_remote`

**Comment by schout-it** [comment]
That is what I did, and after resolving all the rebase conflicts I did a diff to be sure and saw the 2 changes that where not picked up in the rebase

Did it again to pick up the latest changes

**Comment by MiquelRForgeFlow** [comment]
It seems you are not fetching correctly for some reason.

please do:
```
    git remote update
```
and then the rebase

**Comment by schout-it** [comment]
before trying the last time, I deleted the local repo, cloned it back and added oca as a remote.
The rebase always starts back at the beginning
on everythin not related to the changes I made, I do a 
git checkout --ours [path of the file]

It seems some patches will not apply, will need to look at them more closely

File: report_qweb_pdf_watermark/models/report.py [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
+from io import BytesIO
 
-from pyPdf import PdfFileWriter, PdfFileReader
-from pyPdf.utils import PdfReadError
-try:
-    from PyPDF2 import PdfFileWriter, PdfFileReader  # pylint: disable=W0404
-    from PyPDF2.utils import PdfReadError  # pylint: disable=W0404
-except ImportError:
-    pass
+from PyPDF2 import PdfFileWriter, PdfFileReader  # pylint: disable=W0404
```
Comment (khaeusler) [inline report_qweb_pdf_watermark/models/report.py]: the pylint flag for reimported is not necessary

Comment (pedrobaeza) [reply report_qweb_pdf_watermark/models/report.py]: Please don't remove the import protection

**Comment by pedrobaeza** [comment]
Can you squash a bit commit history for merging this?

**Comment by MiquelRForgeFlow** [comment]
I don't understand why commits from other modules are here.

**Comment by pedrobaeza** [comment]
Trying to rebase myself.

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**